src/ssh/cockpitsshrelay.c

```diff
@@ -531,6 +508,29 @@ static void cleanup_knownhosts_file (void)
     }
 }
 
+static gchar *
+challenge_for_knownhosts_data (CockpitSshData *data)
+{
+  const gchar *value = NULL;
+  gchar *ret = NULL;
+  gchar *response = NULL;
+
+  response = challenge_for_auth_data ("x-host-key", data->outfd, NULL);
+  if (response)
+    {
+      value = cockpit_authorize_type (response, NULL);
+      /* Legacy blank string means force fail */
+      if (value && value[0] == '\0')
+        ret = g_strdup ("* invalid key");
+      else
+        ret = g_strdup (value);
+    }
+
+
+  g_free (response);
+  return ret;
+}
```

Comment:
Why was this moved further down? This might just be accidental diff noise?

Reply:
Oh yea, probably noise; sorry will get that.